src/states/common/approved.rs

```diff
@@ -400,8 +415,10 @@ pub trait Approved: Base {
             self, their_pub_id
         );
 
-        self.peer_map_mut().insert(their_pub_id, their_conn_info);
-        self.send_direct_message(&their_pub_id, DirectMessage::ConnectionResponse);
+        self.peer_map_mut()
+            .insert(their_pub_id, their_conn_info.clone());
+        let their_p2p_node = P2pNode::new(their_pub_id, their_conn_info);
+        self.send_direct_message(&their_p2p_node, DirectMessage::ConnectionResponse);
```

Comment:
Again, no need to create the `P2pNode`. Just pass the `ConnectionInfo` directly.